### PR TITLE
Reproject points when navigating to location

### PR DESCRIPTION
**Pull request by markschlosseratbentley**
The globe location tools now will properly use GCS reprojection when navigating. Previously, navigating to certain cartographic locations within the iModel extents could be slightly inaccurate.

The tools affected are:

* ViewGlobeSatelliteTool
* ViewGlobeBirdTool
* ViewGlobeLocationTool
* ViewGlobeIModelTool

The ViewGlobeLocationTool has been further improved to navigate better across long distances when using plane mode.

### CODE REVIEW

File: core/frontend/src/BackgroundMapGeometry.ts [hunk 1]
```diff
@@ -49,12 +49,12 @@ export class BackgroundMapGeometry {
   private static _scratchRayAngles = new Array<LongitudeLatitudeNumber>();
   private static _scratchPoint = Point3d.createZero();
 
-  constructor(private _bimElevationBias: number, globeMode: GlobeMode, iModel: IModelConnection) {
-    this._ecefToDb = iModel.backgroundMapLocation.getMapEcefToDb(_bimElevationBias);
+  constructor(private _bimElevationBias: number, globeMode: GlobeMode, private _iModel: IModelConnection) {
```
Comment (pmconne) [inline core/frontend/src/BackgroundMapGeometry.ts:52]: Why is `_bimElevationBias` now a member? I don't see `this._bimElevationBias` referenced anywhere.

File: core/frontend/src/ViewGlobalLocation.ts [hunk 2]
```diff
@@ -72,19 +72,34 @@ export async function queryTerrainElevationOffset(viewport: ScreenViewport, cart
   return 0;
 }
 
+function _areaToEyeHeight(view3d: ViewState3d, ne?: Point3d, sw?: Point3d, offset = 0): number {
+  if (ne === undefined || sw === undefined)
+    return 0;
+  const diagonal = ne.distance(sw);
+  return diagonal / Math.tan(view3d.camera.getLensAngle().radians / 2.0) + offset;
```
Comment (pmconne) [inline core/frontend/src/ViewGlobalLocation.ts:79]: Check for division by zero?

File: core/frontend/src/ViewState.ts [hunk 3]
```diff
@@ -1300,6 +1346,16 @@ export abstract class ViewState3d extends ViewState {
     return backgroundMapGeometry ? backgroundMapGeometry.cartographicToDb(cartographic, result) : undefined;
   }
 
+  public async rootToCartographicFromGcs(root: XYAndZ, result?: Cartographic): Promise<Cartographic | undefined> {
```
Comment (pmconne) [inline core/frontend/src/ViewState.ts:1349]: These are `@public`. Docs?

File: docs/changehistory/NextVersion.md [hunk 4]
```diff
@@ -16,3 +16,16 @@ ignore: true
 ## Updated version of Electron
 
 Updated version of electron used from 8.2.1 to 10.1.3. Note that Electron is specified as a peer dependency in the iModel.js stack - so it's recommended but not mandatory that applications migrate to this electron version.
+
+## Globe location tool fixes
```
Comment (pmconne) [inline docs/changehistory/NextVersion.md:20]: Mention here the new `FromGcs` versions of various functions?

**Review by pmconne — CHANGES_REQUESTED** [review]

File: core/frontend/src/BackgroundMapGeometry.ts (hunk 1 above)
Comment (markschlosseratbentley) [reply core/frontend/src/BackgroundMapGeometry.ts:52]: @RBBentley - do you know the answer to Paul's queston?

File: core/frontend/src/ViewGlobalLocation.ts (hunk 2 above)
Comment (markschlosseratbentley) [reply core/frontend/src/ViewGlobalLocation.ts]: Added a check.

File: core/frontend/src/BackgroundMapGeometry.ts (hunk 1 above)
Comment (pmconne) [reply core/frontend/src/BackgroundMapGeometry.ts:52]: It's `private`. Unless I'm overlooking some usage of it there's no reason for it to be a member (and then I'm disappointed the compiler doesn't complain about it).

File: docs/changehistory/NextVersion.md (hunk 4 above)
Comment (markschlosseratbentley) [reply docs/changehistory/NextVersion.md:20]: Added

File: core/frontend/src/ViewState.ts (hunk 3 above)
Comment (markschlosseratbentley) [reply core/frontend/src/ViewState.ts:1352]: Added some docs to these; the functions Ray based these on were also not documented so I tried to describe it as best as I understand from the name and the usage.

Comment (pmconne) [reply core/frontend/src/ViewState.ts:1352]: Gonna need a definition of "root space" - is that the same as "spatial" (ask @RBBentley).

File: core/frontend/src/BackgroundMapGeometry.ts (hunk 1 above)
Comment (markschlosseratbentley) [reply core/frontend/src/BackgroundMapGeometry.ts:52]: It seems like it used internally; places like the following:

this._ecefToDb = _iModel.backgroundMapLocation.getMapEcefToDb(_bimElevationBias);

this._mercatorFractionToDb = this._mercatorTilingScheme.computeMercatorFractionToDb(this._ecefToDb, _bimElevationBias, _iModel, false);

Comment (markschlosseratbentley) [reply core/frontend/src/BackgroundMapGeometry.ts:52]: Of course, those are in the constructor, so not necessarily needed to be a member then.

But I do see it used here in a method:

  public getPlane(offset = 0) {
    return Plane3dByOriginAndUnitNormal.create(Point3d.create(0, 0, this._bimElevationBias + offset), Vector3d.create(0, 0, 1))!;
  }

Comment (pmconne) [reply core/frontend/src/BackgroundMapGeometry.ts:52]: I see; it was already a member prior to these changes.
I hate having to consult the constructor to find out what the members of a class are but ok.

**Comment by markschlosseratbentley** [comment]
'rush lint' shows floating promises need some work a few places.

File: core/frontend/src/tools/ViewTool.ts [hunk 5]
```diff
@@ -3233,8 +3233,8 @@ export class ViewGlobeBirdTool extends ViewTool {
       this._beginDoBirdView(viewport, this.oneShot, this.doAnimate);
   }
```
Comment (markschlosseratbentley) [inline core/frontend/src/tools/ViewTool.ts:3235]: 'rush lint' shows floating promises need some work in a few places.

File: core/frontend/src/ViewState.ts (hunk 3 above)
Comment (markschlosseratbentley) [reply core/frontend/src/ViewState.ts:1352]: Ray verified that yes it is spatial space.  I've now documented the functions as such.

File: core/frontend/src/tools/ViewTool.ts (hunk 5 above)
Comment (markschlosseratbentley) [reply core/frontend/src/tools/ViewTool.ts:3243]: Fixed these up.

**Review by pmconne — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Pull request merged**